allensdk/brain_observatory/behavior/data_objects/rewards.py

```diff
@@ -0,0 +1,81 @@
+import pandas as pd
+from pynwb import NWBFile, TimeSeries, ProcessingModule
+
+from allensdk.brain_observatory.behavior.data_files import StimulusFile
+from allensdk.brain_observatory.behavior.data_objects import DataObject, \
+    StimulusTimestamps
+from allensdk.brain_observatory.behavior.data_objects.base \
+    .readable_interfaces import \
+    StimulusFileReadableInterface, NwbReadableInterface
+from allensdk.brain_observatory.behavior.data_objects.base\
+    .writable_interfaces import \
+    NwbWritableInterface
+
+
+class Rewards(DataObject, StimulusFileReadableInterface, NwbReadableInterface,
+              NwbWritableInterface):
+    def __init__(self, rewards: pd.DataFrame):
+        super().__init__(name='rewards', value=rewards)
+
+    @classmethod
+    def from_stimulus_file(cls, stimulus_file: StimulusFile) -> "Rewards":
+        """Get reward data from pkl file, based on pkl file timestamps
+        (not sync file).
+        """
+        data = stimulus_file.data
+        timestamps = StimulusTimestamps.from_stimulus_file(
```

Comment:
This looks like it would be fine for Behavior only Rewards. But, won't this do the wrong thing for BehaviorOphys Rewards? In the case where you have sync data, you should always try to get timestamps for that over the stimulus *.pkl file?